src/Native/System.Security.Cryptography.Native/pal_bio.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_bio.h"
+
+#include <assert.h>
+
+extern "C" BIO* CreateMemoryBio()
+{
+    return BIO_new(BIO_s_mem());
+}
+
+extern "C" BIO* BioNewFile(const char* filename, const char* mode)
+{
+    return BIO_new_file(filename, mode);
+}
+
+extern "C" int32_t BioDestroy(BIO* a)
+{
+    return BIO_free(a);
+}
+
+extern "C" int32_t BioGets(BIO* b, char* buf, int32_t size)
+{
+    return BIO_gets(b, buf, size);
+}
+
+extern "C" int32_t BioRead(BIO* b, void* buf, int32_t len)
+{
+    return BIO_read(b, buf, len);
+}
+
+extern "C" int32_t BioWrite(BIO* b, const void* buf, int32_t len)
+{
+    return BIO_write(b, buf, len);
+}
+
+extern "C" int32_t GetMemoryBioSize(BIO* bio)
+{
+    long ret = BIO_get_mem_data(bio, nullptr);
+    
+    assert(ret <= INT32_MAX);
```

Comment:
BIO_get_mem_data is a macro over top of `BIO_ctrl(b,BIO_CTRL_INFO,0,(char *)pp)`.  When BIO_ctrl called with BIO_CTRL_INFO, it returns 
```C++
case BIO_CTRL_INFO:
        ret = (long)ctx->cont;
        break;
```
where ctx->cont is an int.

Also, all the callers of GetMemoryBioSize use the result for the capacity to create arrays and StringBuilders, so it needs to get casted to an int32 somewhere.

I've added a comment.